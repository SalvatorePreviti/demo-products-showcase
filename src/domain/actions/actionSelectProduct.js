import { SELECT_PRODUCT } from '../actionTypes'

export default function selectProduct(product) {
  return { type: SELECT_PRODUCT, product }
}
