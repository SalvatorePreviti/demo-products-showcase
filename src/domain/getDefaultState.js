function getDefaultState() {
  return {
    categories: [],
    activeCategory: null,
    productFilter: '',
    filteredProducts: []
  }
}

export default getDefaultState
