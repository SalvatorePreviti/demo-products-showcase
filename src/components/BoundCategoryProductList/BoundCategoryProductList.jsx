import { connect } from 'react-redux'
import ProductList from '../../atoms/ProductList/ProductList'
import actionSelectProduct from '../../domain/actions/actionSelectProduct'

function mapStateToProps(state) {
  const { filteredProducts } = state
  return {
    items: filteredProducts
  }
}

function mapDispatchToProps(dispatch) {
  return {
    onSelect(product) {
      dispatch(actionSelectProduct(product))
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProductList)
