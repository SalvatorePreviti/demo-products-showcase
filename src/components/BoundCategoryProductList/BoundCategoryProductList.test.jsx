import React from 'react'
import getDefaultState from '../../domain/getDefaultState'
import mockStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import BoundCategoryProductList from './BoundCategoryProductList'
import actionSelectProduct from '../../domain/actions/actionSelectProduct'

describe('BoundCategoryProductList', () => {
  const items = [{ id: '1', title: 'title1', active: false }, { id: '2', title: 'title2', active: false }]

  let root
  let store

  beforeAll(() => {
    store = mockStore()({
      ...getDefaultState(),
      filteredProducts: items
    })
    root = shallow(<BoundCategoryProductList store={store} />)
  })

  it('snapshots', () => {
    expect(root).toMatchSnapshot()
  })

  it('passes props', () => {
    expect(root.props()).toMatchObject({ items })
  })

  it('passes callbacks', () => {
    root.simulate('select', items[1])
    expect(store.getActions()).toEqual([actionSelectProduct(items[1])])
  })
})
