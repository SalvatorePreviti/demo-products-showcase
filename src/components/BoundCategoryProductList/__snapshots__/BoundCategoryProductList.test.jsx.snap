// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoundCategoryProductList snapshots 1`] = `
<ProductList
  items={
    Array [
      Object {
        "active": false,
        "id": "1",
        "title": "title1",
      },
      Object {
        "active": false,
        "id": "2",
        "title": "title2",
      },
    ]
  }
  onSelect={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
